web/src/components/Presentation/AttributeItem.vue

```diff
@@ -2,7 +2,63 @@
 import { defineComponent, PropType } from '@vue/composition-api';
 import { getField } from '@/encoding';
 import { fieldDisplayName, valueDisplayName } from '@/util';
-import { BaseSearchResult, BiosampleSearchResult } from '@/data/api';
+import { BaseSearchResult, BiosampleSearchResult, JSONValue } from '@/data/api';
+
+export enum Format {
+  RangeWithUnit = 'RangeWithUnit',
+  Range = 'Range',
+  Unit = 'Unit',
+}
+
+/**
+ * Helper function that returns a string representation of the depth in the specified format, if the specified
+ * depth annotation contains adequate information to do so. Otherwise, the function returns `undefined`.
+ */
+const buildStrFromDepthAnnotation = (depthAnnotation: JSONValue, format: Format): string | undefined => {
+  let str: string | undefined;
+
+  // Regardless of the specified format, check whether the depth annotation is a "non-null, non-array" object.
+  if (typeof depthAnnotation === 'object' && depthAnnotation !== null && !Array.isArray(depthAnnotation)) {
+    // Extract data according to the specified format.
+    switch (format) {
+      case Format.RangeWithUnit: {
+        const { has_minimum_numeric_value, has_maximum_numeric_value, has_unit } = depthAnnotation;
+        if (typeof has_minimum_numeric_value === 'number' && typeof has_maximum_numeric_value === 'number' && typeof has_unit === 'string') {
```

Comment:
```suggestion
        if (has_minimum_numeric_value && has_maximum_numeric_value && has_unit) {
```

Could existence checks be sufficient here?

Reply:
Thanks for bringing this up. I don't know.

The reason I did a type check is that I only wanted to use the value if it had the type I specified; and I didn't know what possible types the value might have. I tried to find that out by looking at the SQLAlchemy migrations and TypeScript type definitions, but failed to find type information for the `depth` annotation.

- The [SQL migrations](https://github.com/microbiomedata/nmdc-server/blob/e301cf26c00be161f0ee31ff1087f86c743161a5/nmdc_server/migrations/versions/780ab86a4f27_squashed_initial.py#L193) describe the `annotations` object as arbitrary JSON.
- The [TypeScript type definitions](https://github.com/microbiomedata/nmdc-server/blob/e301cf26c00be161f0ee31ff1087f86c743161a5/web/src/data/api.ts#L47) describe the `annotations` object as one whose properties contain either strings or arrays of strings (which I thought was incorrect, as the `depth` annotation is neither a string nor an array of strings).

For that reason, I thought checking for truthiness might be insufficient, since the property might contain something like an array of strings.

Also, I didn't want to give special treatment to the number `0` (I assumed `0` was a valid depth), and JavaScript treats `0` as falsy.

Reply:
Resolved. In a subsequent commit, I documented my rationale for checking types.

In case that's insufficient, please add a new comment (I don't want to add any code you aren't comfortable with having in the repo).